### DISCUSSION TITLE
Vídeo 08 - Desenvolvimento

### DISCUSSION THREAD

**Discussion by thpoiani** [thpoiani/lab-quarkus#11] [General]
Nesse vídeo falamos rapidamente sobre Programação Reativa.
A arquitetura do Quarkus é fortemente inspirada nesse paradigma.

Você conhece outras bibliotecas para programação reativa? [ReactiveX (RxJava)](https://reactivex.io/), [Project Reactor](https://projectreactor.io/), [Vert.X](https://vertx.io/), **[Mutiny](https://smallrye.io/smallrye-mutiny)**, ...

---

⭐ Fique atento ao **Project Loom**: https://blogs.oracle.com/javamagazine/post/java-loom-virtual-threads-platform-threads

> Project Loom é uma iniciativa para melhorar o desempenho e a escalabilidade do processamento concorrente em Java. O objetivo principal do projeto é tornar mais fácil e eficiente escrever código concorrente, especialmente para aplicações com muitas threads, implementando uma nova forma de lidar com as threads chamada de **"Virtual Threads"**.

Possivelmente o Project Loom estará disponível na próxima **versão LTS de Java (21)**, com planejamento de release para **Setembro de 2023**.

🔴 Fique atento, pois essa feature tem potencial para matar programação reativa!
Sugiro assistir essa entrevista com @briangoetz, um dos arquitetos da linguagem Java que está liderando evoluções na linguagem: https://www.youtube.com/watch?v=9si7gK94gLo&t=1156

**Comment by didifive** [comment]
Eu aprendi sobre programação reativa no Java com o Spring WebFlux, creio que um dos desafios para entender e aproveitar do melhor que a programação reativa tem para oferecer é fazer a desconstrução de execuções síncronas para assíncronas.
Pelo que pesquisei o Java 21 já possui o recurso de Virtual Threads, nos docs do Quarkus já tem o assunto: https://quarkus.io/guides/virtual-threads
